src/HotChocolate/Core/src/Execution/Processing/OperationContextPoolPolicy.cs

```diff
@@ -22,12 +21,20 @@ public bool Return(OperationContext obj)
                 return true;
             }
 
-            if (obj.Execution.TaskStats.IsCompleted)
+            // if work related to the operation context has completed or if it never started
+            // we can reuse the operation context.
+            if (obj.Execution.TaskStats.IsCompleted ||
```

Comment:
The case for reusing a context is now broader. Meaning if we have an early abort by a query, mutation or subscription we also will reuse the context.